### PR TITLE
Audit and Fix `Logic.md`

### LINKED ISSUE #52: Audit and Fix logic.md
This issue is a sub-issue of #18.

TO-DO 
----

- [x] Look for any typos
- [x] Make sure all Markdown is correctly done. The conversion was done automatically by a tool, so there may be more idiomatic ways to do some things
- [x] Check the page in a Desktop browser. Does it look OK? Are there any issues that need to be fixed?
- [x] Check the page in a Mobile browser. Does it look OK? Are there any issues that need to be fixed?
- [x] Try the page in both Light and Dark mode. Are there any issues that need to be fixed?
- [x] Try running the page through Lighthouse: https://developers.google.com/web/tools/lighthouse. Are there any issues that need to be fixed?
- [x] Try running the page through Web Hint: https://webhint.io/. Are there any issues that need to be fixed?
- [x] Tables that are really wide (types.md) could be converted to use SVG or flex/grid. Make wide visual tables fit in the available viewport. Another example is the table in docs/D-Modularity/pointers.md, where we should reduce the vertical padding/margins on tables. The tables are too large, and probably shouldn't be tables at all.
- [x] Prefer Markdown tables vs. raw HTML tags
- [x] Fix colours in tables so it uses global styles vs. inline styles
- [x] Use alt tags for all images. See https://supercooldesign.co.uk/blog/how-to-write-good-alt-text for suggestions
- [x] Avoid blockquote for indentation, and prefer q or blockquote only when actually quoting. Use Admonitions instead. Example: docs/D-Modulatirty/pointers.md: there are places where > Blockquote are being used and Admonitions could replace it. “Note:” Lightbulb info Admonition.
- [x] Make sure images work in dark and light modes. For example, docs/8-Computations/logic.md Flags image is white background, needs to work in dark. Create a “negative” or SVG so we can do Dark Mode? Can we programmatically create this negative?
- [x] Change colours for tables (don’t use Yellow). We should use the default colour theme.
- [x] Avoid using \<b> and \<i> to do emphasis.
- [x] Use backticks and all code for memory addresses and such in tables that use things like void, int, function names, memory addresses.
- [x] Make sure all code blocks use syntax highlighting
- [x] centre images horizontally (esp. when they are small) so they work well at all responsive sizes
- [x] check sub-script and super-script that they are correctly used in all cases
- [x] consider including inter-site links to other pages that have related info
- [x] all links should have descriptive text for accessibility
- [x] Make sure that sub-headings (right-hand side) for each document are correct.
- [x] Improve readability of code comments (i.e., // and /*...*/) in dark mode (even light mode might need some work)
- [x] In places where ***...*** was used as a fourth heading level, use #### instead.
- [x] Fix Frontmatter for page to include proper id, title, slug, etc.

### PR DESCRIPTION
**Pull request by Reza9472**
Fixes #52 
---
Remove redundant _****_

Change images to transparent background

Fix frontmatter page

Center Images

### CODE REVIEW

**Review by MizuhoOkimoto — CHANGES_REQUESTED** [review]
Hi Reza! That's great work you did here! I found some extra spaces on the md file. Is it possible to remove them for improving readability on md file?
I've attached an image below to show which space I'm referring to. (There are more extra spaces in other paragraphs too)
image

**Review by CameronGray1210 — CHANGES_REQUESTED** [review]
Images in dark mode work best when a background (light colour/white) is used and this will work for both modes (light and dark).  Let's not have transparent/alpha channels applied to the images.

File: docs/B-Computations/logic.md [hunk 1]
```diff
@@ -14,7 +18,7 @@ After reading this section, you will be able to:
 
 A complete programming language includes facilities to implement sequential constructs, in which one statement follows another and the statements are executed in order, and two other constructs, which represent modifications of sequential constructs.  Selection constructs represent different paths through the set of instructions.  Iteration constructs represent repetition of the same set of instructions until a specified condition has been met.  The three classes of constructs required to complete a programming language are illustrated in the figure below.
 
-![](https://ict.senecacollege.ca//~ipc144/pages/images/constructs.png)
+![constructs](https://ict.senecacollege.ca//~ipc144/pages/images/constructs.png)
```
Comment (abatomunkuev) [inline docs/B-Computations/logic.md:21]: Suggestion to provide meaningful alt text. Example: `Diagrams of three classes of constructs`

https://supercooldesign.co.uk/blog/how-to-write-good-alt-text

File: docs/B-Computations/logic.md [hunk 2]
```diff
@@ -24,7 +28,7 @@ This chapter introduces the selection and iteration constructs supported by the
 
 A structured program consists of sets of simple constructs, each of which has one entry point and one exit point.  Any programmer may replace one construct with an upgraded construct without affecting the other constructs in the program or introducing errors \("bugs"\). 
 
-![](https://ict.senecacollege.ca//~ipc144/pages/images/structured.png)
+![structured](../../static/img/image39.png)
```
Comment (abatomunkuev) [inline docs/B-Computations/logic.md:31]: Suggestion to provide meaningful alt text. Example: `Diagram of a complete program consisting of multiple replaceable constructs`

File: docs/B-Computations/logic.md [hunk 3]
```diff
@@ -69,7 +73,7 @@ _**Pseudo-code**_ is a set of shorthand notes in a human \(non-programming\) lan
 
 A _**flow chart**_ is a set of conventional symbols connected by arrows that illustrate the flow of control through a programming solution.  Popular sets of symbols for sequences, selections and iterations are shown below:
 
-![](https://ict.senecacollege.ca//~ipc144/pages/images/flowChart.png)
+![flowChart](../../static/img/image40.png)
```
Comment (abatomunkuev) [inline docs/B-Computations/logic.md:76]: Suggestion to provide meaningful alt text. Example: `Popular symbols for sequence, selection, and iteration constructs`

File: docs/B-Computations/logic.md [hunk 4]
```diff
@@ -83,7 +87,7 @@ The C language supports three selection constructs:
 
 The flow charts for these three constructs are shown below:
 
-![](https://ict.senecacollege.ca//~ipc144/pages/images/selection.png)
+![selection](../../static/img/image42.png)
```
Comment (abatomunkuev) [inline docs/B-Computations/logic.md:90]: Suggestion to provide meaningful alt text. Example: `Flow charts of the three selection constructs`

File: docs/B-Computations/logic.md [hunk 5]
```diff
@@ -283,7 +287,7 @@ The C language supports three iteration constructs:
 
 Three instructions control the execution of an iteration: an initialization, a test condition, a change statement.  The test condition may be simple or compound.  The flow charts for the three constructs are shown below.
 
-![](https://ict.senecacollege.ca//~ipc144/pages/images/iteration.png)
+![iteration](../../static/img/image38.png)
```
Comment (abatomunkuev) [inline docs/B-Computations/logic.md:290]: Suggestion to provide meaningful alt text. Example: `Flow charts of the three iteration constructs`

File: docs/B-Computations/logic.md [hunk 6]
```diff
@@ -391,7 +395,8 @@ for (slices = 4; slices > 0; --slices)
 
 Flagging is a method of coding iteration constructs within the single-entry single-exit rule of structured programming.  Consider the flow-chart on the left side in the figure below.  This design contains a path that crosses another path. 
 
-![](https://ict.senecacollege.ca//~ipc144/pages/images/flags.png)
+![flags](/img/image37.png)
```
Comment (abatomunkuev) [inline docs/B-Computations/logic.md:398]: Suggestion to provide meaningful alt text. Example: `Spaghetti code versus Structured code flow charts`

**Review by abatomunkuev — COMMENTED** [review]
Hello! I have reviewed the changes related to Markdown file. Consider the provided comments.

File: docs/B-Computations/logic.md (hunk 1 above)
Comment (Reza9472) [reply docs/B-Computations/logic.md:21]: Amazing work!
Thank you @abatomunkuev !

**Comment by Reza9472** [comment]
> Hi Reza! That's great work you did here! I found some extra spaces on the md file. Is it possible to remove them for improving readability on md file? I've attached an image below to show which space I'm referring to. (There are more extra spaces in other paragraphs too) image

@MizuhoOkimoto Good catch! Thank you

**Comment by Reza9472** [comment]
@CameronGray1210 Thanks for the comment. I have updated my code.
Please take a look again

**Review by MizuhoOkimoto — APPROVED** [review]
Looks good!

**Review by humphd — CHANGES_REQUESTED** [review]
Waiting on rebase.

**Comment by Reza9472** [comment]
> Waiting on rebase.

@humphd Thank you. I've updated the custom.css

**Comment by Reza9472** [comment]
@humphd Conflicts Resolved. Thank you

File: docs/B-Computations/logic.md [hunk 7]
```diff
@@ -14,19 +18,19 @@ After reading this section, you will be able to:
...
 
 This chapter introduces the selection and iteration constructs supported by the C language and describes how to implement structured programming principles in coding iterations.
 
 ## Structured Programming
 
-A structured program consists of sets of simple constructs, each of which has one entry point and one exit point.  Any programmer may replace one construct with an upgraded construct without affecting the other constructs in the program or introducing errors \("bugs"\). 
+A structured program consists of sets of simple constructs, each of which has one entry point and one exit point. Any programmer may replace one construct with an upgraded construct without affecting the other constructs in the program or introducing errors \("bugs"\). 
 
-![](https://ict.senecacollege.ca//~ipc144/pages/images/structured.png)
+![Diagram of a complete program consisting of multiple replaceable constructs](https://ict.senecacollege.ca//~ipc144/pages/images/structured.png)
```
Comment (humphd) [inline docs/B-Computations/logic.md:31]: This image needs to be put in the repo, not linked to remotely.

File: docs/B-Computations/logic.md [hunk 8]
```diff
@@ -69,7 +73,7 @@ _**Pseudo-code**_ is a set of shorthand notes in a human \(non-programming\) lan
 
 A _**flow chart**_ is a set of conventional symbols connected by arrows that illustrate the flow of control through a programming solution.  Popular sets of symbols for sequences, selections and iterations are shown below:
 
-![](https://ict.senecacollege.ca//~ipc144/pages/images/flowChart.png)
+![Popular symbols for sequence, selection, and iteration constructs](https://ict.senecacollege.ca//~ipc144/pages/images/flowChart.png)
```
Comment (humphd) [inline docs/B-Computations/logic.md:76]: Same issue here.

File: docs/B-Computations/logic.md [hunk 9]
```diff
@@ -83,7 +87,7 @@ The C language supports three selection constructs:
 
 The flow charts for these three constructs are shown below:
 
-![](https://ict.senecacollege.ca//~ipc144/pages/images/selection.png)
+![Flow charts of the three selection constructs](https://ict.senecacollege.ca//~ipc144/pages/images/selection.png)
```
Comment (humphd) [inline docs/B-Computations/logic.md:90]: And here

File: docs/B-Computations/logic.md [hunk 10]
```diff
@@ -281,15 +285,15 @@ The C language supports three iteration constructs:
 * do while
 * for
 
-Three instructions control the execution of an iteration: an initialization, a test condition, a change statement.  The test condition may be simple or compound.  The flow charts for the three constructs are shown below.
+Three instructions control the execution of an iteration: an initialization, a test condition, a change statement. The test condition may be simple or compound.  The flow charts for the three constructs are shown below.
 
-![](https://ict.senecacollege.ca//~ipc144/pages/images/iteration.png)
+![Flow charts of the three iteration constructs](https://ict.senecacollege.ca//~ipc144/pages/images/iteration.png)
```
Comment (humphd) [inline docs/B-Computations/logic.md:290]: And here

File: docs/B-Computations/logic.md [hunk 11]
```diff
@@ -389,11 +393,12 @@ for (slices = 4; slices > 0; --slices)
 
 ## Flags
 
-Flagging is a method of coding iteration constructs within the single-entry single-exit rule of structured programming.  Consider the flow-chart on the left side in the figure below.  This design contains a path that crosses another path. 
+Flagging is a method of coding iteration constructs within the single-entry single-exit rule of structured programming.  Consider the flow-chart on the left side in the figure below. This design contains a path that crosses another path. 
 
-![](https://ict.senecacollege.ca//~ipc144/pages/images/flags.png)
+![Spaghetti code versus Structured code flow charts](https://ict.senecacollege.ca//~ipc144/pages/images/flags.png)
```
Comment (humphd) [inline docs/B-Computations/logic.md:398]: And here

**Review by humphd — CHANGES_REQUESTED** [review]
Revert the changes to `package.json` and `src/css/custom.css`:

```
git checkout main package.json
git checkout main src/css/custom.css
```

**Comment by Reza9472** [comment]
@humphd Done. Please review

**Review by humphd — CHANGES_REQUESTED** [review]
Looks pretty good, but you don't need the changes to `package.json` or the lock file:

```
git checkout package.json main
git checkout package-lock.json main
```

**Review by humphd — APPROVED** [review]

**Pull request merged**